### PR TITLE
Quick & dirty metric for background update status

### PR DESCRIPTION
**Pull request by DMRobertson**
Tested with the demo scripts briefly:

```
$ curl -s http://localhost:8080/_synapse/metrics | grep background_update_status
# HELP synapse_background_update_status Background update status
# TYPE synapse_background_update_status gauge
synapse_background_update_status 3.0
```

### CODE REVIEW

File: synapse/storage/background_updates.py [hunk 1]
```diff
@@ -158,11 +168,16 @@ def __init__(self, hs: "HomeServer", database: "DatabasePool"):
...
         self._background_update_handlers: Dict[str, _BackgroundUpdateHandler] = {}
+        # TODO: all these bool flags make me feel icky---can we combine into a status
+        # enum?
         self._all_done = False
 
         # Whether we're currently running updates
         self._running = False
 
+        # Marker to be set if we abort and halt all background updates.
+        self._aborted = False
```
Comment (MatMaul) [inline synapse/storage/background_updates.py:179]: This var seems to never be written for now, is it a provision for future use ?

File: synapse/storage/background_updates.py [hunk 2]
```diff
@@ -175,6 +190,20 @@ def __init__(self, hs: "HomeServer", database: "DatabasePool"):
         self.sleep_duration_ms = hs.config.background_updates.sleep_duration_ms
         self.sleep_enabled = hs.config.background_updates.sleep_enabled
 
+    def get_status(self) -> UpdaterStatus:
+        """An integer summarising the updater status. Used as a metric."""
+        logger.warning("DMR: giff status pls")
```
Comment (MatMaul) [inline synapse/storage/background_updates.py:195]:
```suggestion
```
:angel:

**Review by MatMaul — CHANGES_REQUESTED** [review]

File: synapse/storage/background_updates.py (hunk 1 above)
Comment (DMRobertson) [reply synapse/storage/background_updates.py:179]: OOOOOPS.

**Comment by DMRobertson** [comment]
Maybe I should write some tests for this...

File: synapse/storage/background_updates.py [hunk 3]
```diff
@@ -158,11 +168,16 @@ def __init__(self, hs: "HomeServer", database: "DatabasePool"):
 
         self._background_update_performance: Dict[str, BackgroundUpdatePerformance] = {}
         self._background_update_handlers: Dict[str, _BackgroundUpdateHandler] = {}
+        # TODO: all these bool flags make me feel icky---can we combine into a status
```
Comment (MatMaul) [inline synapse/storage/background_updates.py:171]: Do you see any trouble with doing so by using the enum you are creating ? It seems a small enough change to be included in this PR.

Comment (DMRobertson) [reply synapse/storage/background_updates.py:171]: I just didn't want to dive into the machinery behind the background updater here---this was meant to be quick and dirty to get a useful metric out.

**Pull request merged**